docs/developer/ceed.md

```diff
@@ -0,0 +1,206 @@
+| ---------------- | ----------------- | -------------------- | ---------------------------------- | ----- |
+| `restrict_geom`  |  `nOwnedInEdges`  | `4 * nOwnedInEdges`  | `CeedElemRestrictionCreateStrided` |  offset = [ 0,  4,  8, 12, 16, 20, 24] |
+| `q_restrict_l`   |  `nOwnedInEdges`  | `3 * nCells`         | `CeedElemRestrictionCreate`        |  offset = [ 0,  3,  0,  3,  6,  9, 12] |
+| `q_restrict_r`   |  `nOwnedInEdges`  | `3 * nCells`         | `CeedElemRestrictionCreate`        |  offset = [ 3,  6,  9, 12, 15, 12, 15] |
+| `c_restrict_l`   |  `nOwnedInEdges`  | `3 * nCells`         | `CeedElemRestrictionCreate`        |  offset = [ 0,  3,  0,  3,  6,  9, 12] |
+| `c_restrict_r`   |  `nOwnedInEdges`  | `3 * nCells`         | `CeedElemRestrictionCreate`        |  offset = [ 3,  6,  9, 12, 15, 12, 15] |
+| `restrict_flux`  |  `nOwnedInEdges`  | `3 * nOwnedInEdges`  | `CeedElemRestrictionCreateStrided` |  offset = [ 0,  3,  6,  9, 12, 15, 18] |
+
+where `nOwnedInEdges` is the number of owned internal edges and `nCells` is the total number of cells ( = owned + ghost).
+
+- Third,  create the `CeedOperator` using the previously created `CeedQFunction` and all `CeedElementRestriction`.
+The multiple fields are added via `CeedOperatorSetField`.
+
+| Field name   | CeedElemRestriction | CeedVector           | Notes |
+| ------------ | ------------------- | -------------------- | ----- |
+| `geom`       |  `restrict_geom`    | `geom`               | `geom` has values `geom[:][0:3] = [sn cn L/A_l L/A_r]` |
+| `q_left`     |  `q_restrict_l`     | `CEED_VECTOR_ACTIVE` |  |
+| `q_right`    |  `q_restrict_r`     | `CEED_VECTOR_ACTIVE` |  |
+| `cell_left`  |  `c_restrict_l`     | `CEED_VECTOR_ACTIVE` |  |
+| `cell_right` |  `c_restrict_r`     | `CEED_VECTOR_ACTIVE` |  |
+| `flux`       |  `restrict_flux`    | `flux`               | `flux` is initalized to `0.0` |
+
+### `CeedOperator` for Boundary Edges
+
+The values right of the edge are provided if a Dirichlet boundary condition is applied on that edge
+For a reflective boundary condition, the value on the left of the edge is used as the value on right of the edge.
+Based on the mesh shown above, the boundary edges are listed in the table below.
+
+| Boundary Edge | Left Cell | (Optional) Dirichlet Right cell |
+| ---- | ---- | ---- |
+| e00  | c00  | d00  |
+| e01  | c01  | d01  |
+| e02  | c02  | d02  |
+| e03  | c00  | d03  |
+| e06  | c02  | d04  |
+| e10  | c03  | d05  |
+| e13  | c05  | d06  |
+| e14  | c03  | d07  |
+| e15  | c04  | d08  |
+| e16  | c05  | d09  |
+
+The fields for the boundary condition `CeedQFunction` are listed below. It should be noted that the `geom` field for a boundary edge has one geometric attribute less than the `geom` field for an intenral edge because the value `L_edge/A_right` is not needed for a boundary edge. Similarly, `cell_right` field is omitted for a boundary edge.
```

Comment:
one geometric attribute less than the `geom` field for an intenral edge -> one geometric attribute fewer than the `geom` field for an internal edge